test/redundant_rbs_selector_task_test.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+using_task_library "transforms"
+
+describe OroGen.transforms.RedundantRBSSelectorTask do
+    run_live
+
+    attr_reader :task
+
+    before do
+        @task = syskit_deploy(
+            OroGen.transforms.RedundantRBSSelectorTask
+                  .deployed_as("test_task")
+        )
+
+        @task.properties.source_timeout = Time.at(1)
+        @task.properties.main_source_histeresys = Time.at(5)
+
+        syskit_configure(@task)
+    end
+
+    it "forwards main source when both are available" do
+        syskit_start(task)
+
+        main_w = syskit_create_writer task.main_rbs_source_port
+        secondary_w = syskit_create_writer task.secondary_rbs_source_port
+
+        main_rbs = rbs({ data: [1, 0, 0] })
+        secondary_rbs = rbs({ data: [-1, 0, 0] })
+        out = expect_execution.poll do
+            main_w.write main_rbs
+            secondary_w.write secondary_rbs
+        end.to do # rubocop:disable Style/MultilineBlockChain
+            custom_runtime_states.each do |state|
+                not_emit task.send("#{state}_event")
+            end
+            have_new_samples(task.rbs_out_port, 20)
+        end
+
+        out.each do |s|
+            assert_equal 1, s.position[0]
+        end
+    end
+
+    describe "transitions from MAIN_SOURCE_RECOVERING" do
+        it "transition from MAIN_SOURCE_RECOVERING to NO_VALID_SOURCES" do
+            syskit_start(task)
+            transition_to_main_source_recovering
+
+            expect_execution.to do
+                emit task.no_valid_sources_event
+                custom_runtime_states(except: ["no_valid_sources"]).each do |state|
+                    not_emit task.send("#{state}_event")
+                end
```

Comment:
I tried writing a helper like
```suggestion
                emit_only "no_valid_source"
```
but struggled with context and/or expectations parsing (I am not sure realy)

Reply:
You would need to expand an intenral context from Roby/Syskit ... lots of complications for little value.